app/models/comment.rb

```diff
@@ -1,4 +1,7 @@
 class Comment < ActiveRecord::Base
+
+  validates :text, presence: true, unless: :is_supervisor_check?
```

Comment:
I don't know what side-effects this may cause. Sure, it only fires when the `is_supervisor_check?` is true, but I feel like validating on the two dimensions "by whom" (e.g. supervisor, public user, …) and "what kind of commentable" violates the idea of polymorphic commentable.

An idea would be to have something like a StatusUpdateComment [form object](https://medium.com/@ryakh/rails-form-objects-84b6849c886e) that encapsulates the validation, but I think that's too much for this PR.

It's not ideal, either, but I would revert to the `comment.text.present? && coment.save` check for the time being and remove the validation.